Patch-Source: https://sources.debian.org/src/postgresql-14/14.0-1/debian/patches/libpgport-pkglibdir
Author: Christoph Berg <myon@debian.org>
Description: Move libpgport/libpgcommon/libpgfeutils from libdir to pkglibdir
 This allows client applications to link to version-specific libraries.
 Used by pg-checksums.

--- a/src/common/Makefile
+++ b/src/common/Makefile
@@ -68,15 +68,15 @@ all: libpgcommon.a libpgcommon_shlib.a l
 
 # libpgcommon is needed by some contrib
 install: all installdirs
-	$(INSTALL_STLIB) libpgcommon.a '$(DESTDIR)$(libdir)/libpgcommon.a'
-	$(INSTALL_STLIB) libpgcommon_shlib.a '$(DESTDIR)$(libdir)/libpgcommon_shlib.a'
+	$(INSTALL_STLIB) libpgcommon.a '$(DESTDIR)$(pkglibdir)/libpgcommon.a'
+	$(INSTALL_STLIB) libpgcommon_shlib.a '$(DESTDIR)$(pkglibdir)/libpgcommon_shlib.a'
 
 installdirs:
-	$(MKDIR_P) '$(DESTDIR)$(libdir)'
+	$(MKDIR_P) '$(DESTDIR)$(pkglibdir)'
 
 uninstall:
-	rm -f '$(DESTDIR)$(libdir)/libpgcommon.a'
-	rm -f '$(DESTDIR)$(libdir)/libpgcommon_shlib.a'
+	rm -f '$(DESTDIR)$(pkglibdir)/libpgcommon.a'
+	rm -f '$(DESTDIR)$(pkglibdir)/libpgcommon_shlib.a'
 
 libpgcommon.a: $(OBJS_FRONTEND)
 	rm -f $@
--- a/src/fe_utils/Makefile
+++ b/src/fe_utils/Makefile
@@ -35,13 +35,13 @@ distprep: psqlscan.c
 
 # libpgfeutils could be useful to contrib, so install it
 install: all installdirs
-	$(INSTALL_STLIB) libpgfeutils.a '$(DESTDIR)$(libdir)/libpgfeutils.a'
+	$(INSTALL_STLIB) libpgfeutils.a '$(DESTDIR)$(pkglibdir)/libpgfeutils.a'
 
 installdirs:
-	$(MKDIR_P) '$(DESTDIR)$(libdir)'
+	$(MKDIR_P) '$(DESTDIR)$(pkglibdir)'
 
 uninstall:
-	rm -f '$(DESTDIR)$(libdir)/libpgfeutils.a'
+	rm -f '$(DESTDIR)$(pkglibdir)/libpgfeutils.a'
 
 clean distclean:
 	rm -f libpgfeutils.a $(OBJS) lex.backup
--- a/src/port/Makefile
+++ b/src/port/Makefile
@@ -54,15 +54,15 @@ all: libpgport.a libpgport_shlib.a libpg
 
 # libpgport is needed by some contrib
 install: all installdirs
-	$(INSTALL_STLIB) libpgport.a '$(DESTDIR)$(libdir)/libpgport.a'
-	$(INSTALL_STLIB) libpgport_shlib.a '$(DESTDIR)$(libdir)/libpgport_shlib.a'
+	$(INSTALL_STLIB) libpgport.a '$(DESTDIR)$(pkglibdir)/libpgport.a'
+	$(INSTALL_STLIB) libpgport_shlib.a '$(DESTDIR)$(pkglibdir)/libpgport_shlib.a'
 
 installdirs:
-	$(MKDIR_P) '$(DESTDIR)$(libdir)'
+	$(MKDIR_P) '$(DESTDIR)$(pkglibdir)'
 
 uninstall:
-	rm -f '$(DESTDIR)$(libdir)/libpgport.a'
-	rm -f '$(DESTDIR)$(libdir)/libpgport_shlib.a'
+	rm -f '$(DESTDIR)$(pkglibdir)/libpgport.a'
+	rm -f '$(DESTDIR)$(pkglibdir)/libpgport_shlib.a'
 
 libpgport.a: $(OBJS)
 	rm -f $@
--- a/src/Makefile.global.in
+++ b/src/Makefile.global.in
@@ -599,7 +599,7 @@
 ifeq ($(PORTNAME),darwin)
 libpq_pgport = $(libpq)
 else ifdef PGXS
-libpq_pgport = -L$(libdir) -lpgcommon -lpgport $(libpq)
+libpq_pgport = -L$(pkglibdir) -lpgcommon -lpgport $(libpq)
 else
 libpq_pgport = -L$(top_builddir)/src/common -lpgcommon -L$(top_builddir)/src/port -lpgport $(libpq)
 endif
@@ -610,7 +610,7 @@
 # done if they don't, since they will have satisfied all their references
 # from these libraries.)
 ifdef PGXS
-libpq_pgport_shlib = -L$(libdir) -lpgcommon_shlib -lpgport_shlib $(libpq)
+libpq_pgport_shlib = -L$(pkglibdir) -lpgcommon_shlib -lpgport_shlib $(libpq)
 else
 libpq_pgport_shlib = -L$(top_builddir)/src/common -lpgcommon_shlib -L$(top_builddir)/src/port -lpgport_shlib $(libpq)
 endif
